server/methods/toogleFavorite.js

```diff
@@ -9,6 +9,14 @@ Meteor.methods({
 			});
 		}
 
+		const userSubscription = RocketChat.models.Subscriptions.findOneByRoomIdAndUserId(rid, Meteor.userId());
+		if (!userSubscription) {
+			throw new Meteor.Error('error-invalid-subscription',
+				'Invalid subscription',
```

Comment:
I think a better error would be "you can't favorite" a room you aren't part of... I need to test, but before this change what happens when you try to favorite a room that you aren't part of?

Reply:
The endpoint returns success, but no data is changed. =/

Reply:
This is a good change. Let's keep it how it is but change the `Invalid subscription` to `You must be part of a room to favorite it`?